### PR TITLE
[codex] average Tinker final metrics

### PR DESCRIPTION
**Pull request by RaychevAngel**
## Summary
- summarize `metrics.json` using the mean observed step loss instead of the final batch loss
- recompute `primary_metric` from the averaged validation loss
- keep `metrics.jsonl` as the per-step trace for detailed inspection

## Why
Live 5-step paired Tinker runs on the small web-SFT dataset showed large batch-to-batch loss swings because the dataset is tiny and cycled. Using only the last batch made AutoResearch decisions noisier than the observed trajectory warranted.

## Validation
- python3 -m compileall src
- uv run --with pytest --with anthropic --with langgraph python -m pytest tests/test_tinker_runner.py -q
- uv run --with pytest --with anthropic --with langgraph python -m pytest tests/test_tinker_runner.py tests/test_tinker_api.py tests/test_autoresearch.py -q
- local unpublished stack with #55/#56/#57/#58/#59/#60: compileall + broad non-live suite excluding live Tinker/HF passed (`200 passed, 7 skipped`)

## Live validation
- Before this PR, the 5-step paired run compared `learning_rate=0.0001` against `0.0005` on `Qwen/Qwen3.5-9B`; final-batch scoring saw only a small edge (`0.08615` vs `0.08739`), while the per-step trace showed substantial volatility.
- After this PR, a 5-step live run produced `metrics.json` with `val_loss=11.902454127464443`, exactly matching the mean of the per-step `metrics.jsonl` losses, and `primary_metric=0.07750463517412384`.

### CODE REVIEW

**Comment by RaychevAngel** [comment]
Superseded by consolidated replacement PR(s): #114. This old draft was part of the fragmented Tinker SFT runner stack.
